terraform/environments/data-platform/lambda.tf

```diff
@@ -289,8 +289,34 @@ module "data_product_create_schema_lambda" {
       action        = "lambda:InvokeFunction"
       function_name = "data_product_create_metadata_${local.environment}"
       principal     = "apigateway.amazonaws.com"
-      source_arn    = "arn:aws:execute-api:${local.region}:${local.account_id}:${aws_api_gateway_rest_api.data_platform.id}/*/${aws_api_gateway_method.create_schema_for_data_product_table_name.http_method}${aws_api_gateway_resource.create_schema_for_data_product_table_name.path}"
+      source_arn    = "arn:aws:execute-api:${local.region}:${local.account_id}:${aws_api_gateway_rest_api.data_platform.id}/*/${aws_api_gateway_method.create_schema_for_data_product_table_name.http_method}${aws_api_gateway_resource.schema_for_data_product_table_name.path}"
     }
   }
 
 }
+
+module "get_schema_lambda" {
+  source                         = "github.com/ministryofjustice/modernisation-platform-terraform-lambda-function?ref=a4392c1" # ref for V2.1
+  application_name               = "get_schema"
+  tags                           = local.tags
+  description                    = "Fetch the schema for a table from S3"
+  role_name                      = "get_schema_role_${local.environment}"
+  policy_json                    = data.aws_iam_policy_document.iam_policy_document_for_get_schema_lambda.json
+  function_name                  = "get_schema_${local.environment}"
+  create_role                    = true
+  reserved_concurrent_executions = 1
+
+  image_uri    = "374269020027.dkr.ecr.eu-west-2.amazonaws.com/data-platform-get-schema-lambda-ecr-repo:${local.get_schema_version}"
+  timeout      = 600
+  tracing_mode = "Active"
+  memory_size  = 512
+
+  environment_variables = {
+    LOG_BUCKET          = module.s3-bucket.bucket.id
```

Comment:
These need to reference their respective buckets, can use:
`  environment_variables = merge(local.logger_environment_vars, local.storage_environment_vars)`